src/main/scala/edu/ie3/simona/model/participant/hp/HpPowerLimitFlexModel.scala

```diff
@@ -30,11 +29,46 @@ class HpPowerLimitFlexModel(private val model: HpModel)
         wasRunningLastOp,
       )
 
-    PowerLimitFlexOptions(
-      if (turnOn) model.sRated.toActivePower(model.cosPhiRated) else zeroKW,
-      if (canBeOutOfOperation) zeroKW else model.pRated,
-      if (canOperate) model.pRated else zeroKW,
-    )
+    val (refPower, minPower) = (turnOn, canBeOutOfOperation) match {
```

Comment:
Just a remark on code style here. Maybe one could declare a boolean tuple `val (refOn, minOn)` here, whose values get turned into `model.sRated.toActivePower(model.cosPhiRated)` or `zeroKW` later. This way, one could reduce redundant code a bit.